tests/test_constraints_polars.py

```diff
@@ -149,3 +153,50 @@ def test_polars_linked_parameters(parameters, constraints):
 
     num_entries = len(df)
     assert num_entries == 0
+
+
+@pytest.mark.parametrize(
+    "parameter_names",
+    [
+        [
+            "Temperature",
+            "Solvent_1",
+            "Solvent_2",
+            "Solvent_3",
+            "Fraction_1",
+            "Fraction_2",
+            "Fraction_3",
+        ]
+    ],
+)
+@pytest.mark.parametrize(
+    "constraint_names",
+    [
+        ["Constraint_4"],
+        ["Constraint_12"],
+        ["Constraint_14", "Constraint_8", "Constraint_9"],
+    ],
+)
+def test_polars_product(constraints, parameters):
+    """Test the result of parameter product and filtering."""
+    # Do Polars product
+    ldf = parameter_cartesian_prod_polars(parameters)
+    pol_df = ldf.collect()
+
+    # Do Pandas product
+    pd_df = parameter_cartesian_prod_pandas(parameters)
+
+    # Assert equality of lengths before filtering
+    assert len(pol_df.to_pandas()) == len(
```

Comment:
why not also test for equality here?

Reply:
Added.

Reply:
thx, but I think the previous assert can then go (since implied by the equality check)

Reply:
removed